### PR TITLE
Bug 1300076 - SIGPIPE events kill go process

### PR DESCRIPTION
**Pull request by jwhonce**
Required to merge first https://github.com/projectatomic/docker/pull/70
- Update docker.service file to use forward-journald
- Build forward-journald
- Install forward-journald.man

### CODE REVIEW

**Comment by jwhonce** [comment]
@lsm5 @runcom Please see https://github.com/projectatomic/docker/pull/70 for the source code for forward-journald.  I have included the docker.service updates in this PR, but not the updates to the spec file.  If you could provide guidance on the changes necessary, I can create PR's for the other branches based on your recommendations.  :+1:

File: docker.service [hunk 1]
```diff
@@ -6,20 +6,23 @@ Wants=docker-storage-setup.service
...
           $DOCKER_NETWORK_OPTIONS \
-          $INSECURE_REGISTRY
+          $INSECURE_REGISTRY \
+          2>&1 | /usr/bin/forward-journald -filter docker'
 LimitNOFILE=1048576
 LimitNPROC=1048576
 LimitCORE=infinity
-MountFlags=slave
-TimeoutStartSec=1min
+TimeoutStartSec=0
```
Comment (lsm5) [inline docker.service]: so this means we're not waiting at all for the daemon to start, correct? any reason for the change?

File: docker.service [hunk 2]
```diff
@@ -6,20 +6,23 @@ Wants=docker-storage-setup.service
...
+StandardOutput=null
+StandardError=null
```
Comment (lsm5) [inline docker.service:26]: so iiuc, everything gets sent to forward-journald and nothing shows up here, correct?

**Comment by lsm5** [comment]
@jwhonce I guess we'd want this on all branches, but we could cherry-pick once merged. Let's merge once we have spec file changes too.

@runcom wdyt?

**Comment by runcom** [comment]
I thought this was only for docker 1.9 < go 1.6rc2 @jwhonce right?

@lsm5 we also need to update the spec to build and install the tool as a subpkg correct?

File: docker.service (hunk 1 above)
Comment (jwhonce) [inline docker.service]: Oops, that's me using a different source file than yours.  I'll fix that.

File: docker.service (hunk 2 above)
Comment (jwhonce) [inline docker.service:26]: Correct, this is the "meat" of the fix

**Comment by jwhonce** [comment]
@runcom  We need this for any docker built with < go 1.6rc2

**Comment by runcom** [comment]
@jwhonce right, so just f22 and f23 for now I suppose

**Comment by lsm5** [comment]
lgtm, i wanna merge this, @runcom lgty?

**Comment by lsm5** [comment]
i'll add subpackage in spec file once projectatomic/docker is updated.

**Comment by runcom** [comment]
@lsm5 cool, lgtm

**Comment by lsm5** [comment]
this repo was kinda out of sync from the canonical dist-git repo, I'm cherry-picking and then force pushing over here. Sorry about that :| . Your commits are not lost though :)

**Pull request merged**